python/ray/_private/serialization.py

```diff
@@ -292,7 +292,13 @@ def _deserialize_pickle5_data(
                 gpu_object_manager.fetch_gpu_object(object_id)
             tensors = gpu_object_manager.gpu_object_store.get_gpu_object(object_id)
             ctx.reset_out_of_band_tensors(tensors)
-            gpu_object_manager.gpu_object_store.remove_gpu_object(object_id)
+            gpu_object_store = gpu_object_manager.gpu_object_store
+            # If the GPU object is self-created, it means the transfer is intra-actor.
+            # In this case, we should not remove the GPU object after it is consumed once,
+            # to avoid a KeyError if the GPU object reference is used more than once.
+            # Instead, we should wait for the GC callback to clean it up.
```

Comment:
```suggestion
            # If the GPU object is self-created, it means the transfer is intra-actor.
            # In this case, we should not remove the GPU object after it is consumed once,
            # because the GPU object reference may be used again.
            # Instead, we should wait for the GC callback to clean it up.
```